### DISCUSSION TITLE
Cypress install does not honor CYPRESS_INSTALL_BINARY

### DISCUSSION THREAD

**Discussion by JeeMoe** [cypress-io/cypress#25580] [Questions and Help]
At least it appears that it doesn't.  We do not have access to the public internet, but can install via cypress.zip.   We can unzip and run just fine, but that is not the recommended approach in Cypress advanced documentation.  It says set the CYPRESS_INSTALL_BINARY then "npm install cypress."   But cypress continues to try to connect online and finally times out with a connection error.   We actually need to do this with version 9.2.1, but also no success using the latest 12.4.   TIA

**Comment by portyu9** [comment]
Hi @JeeMoe 

`CYPRESS_INSTALL_BINARY` only controls installation of the **Cypress binary**. It does not make `npm install cypress` itself offline.

Cypress installation has two separate parts:

1. the `cypress` npm package
2. the Cypress application/binary downloaded during `postinstall`

For the binary, a local ZIP is supported:

```bash
CYPRESS_INSTALL_BINARY=/absolute/path/to/cypress.zip \
  npm install cypress@9.2.1
```

But `npm` still needs to obtain `cypress@9.2.1` and all of its npm dependencies. If the timeout is against `registry.npmjs.org`, `CYPRESS_INSTALL_BINARY` is not the relevant setting — you need those npm packages available through an internal registry/cache or transferred into the air-gapped environment.

If the timeout is instead against `download.cypress.io` or `cdn.cypress.io`, then the Cypress postinstall process is not seeing the local binary setting. In that case make sure the environment variable is set on the **same process that runs the install** and use an absolute path.

You can also split the two operations for easier diagnosis:

```bash
CYPRESS_INSTALL_BINARY=0 npm install cypress@9.2.1
```

then, once the npm package is available:

```bash
CYPRESS_INSTALL_BINARY=/absolute/path/to/cypress.zip \
DEBUG=cypress:cli* \
npx cypress install
```

For a completely air-gapped installation you therefore need both artifacts available internally: the npm dependency tree and the matching Cypress ZIP.

Also make sure the ZIP corresponds to the Cypress version/platform you intend to run; for your case that means a 9.2.1 binary if the npm package is `cypress@9.2.1`.

Please mark this answer as accepted if it helped, thank you.
